commands/channel_test.go

```diff
@@ -129,3 +129,150 @@ func (s *MmctlUnitTestSuite) TestSearchChannelCmdF() {
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Create channel with a Name and Display Name but Team returns an error", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", mockChannelName, "")
+		cmd.Flags().String("display_name", mockChannelDisplayName, "")
+
+		err := createChannelCmdF(s.client, cmd, []string{})
+
+		s.Require().Equal(err.Error(), "Team is required")
+		s.Require().Len(printer.GetLines(), 0)
+
+	})
+
+	s.Run("Test creating a channel with a nonexistent team returns an error", func() {
+		printer.Clean()
+
+		teamNameArg := "mockTeamName"
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", mockChannelName, "")
+		cmd.Flags().String("display_name", mockChannelDisplayName, "")
+		cmd.Flags().String("team", teamNameArg, "")
+
+		s.client.
+			EXPECT().
+			GetTeam(teamNameArg, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeamByName(teamNameArg, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		err := createChannelCmdF(s.client, cmd, []string{})
+
+		s.Require().Equal(err.Error(), "Unable to find team: "+ teamNameArg)
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Create open channel", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", mockChannelName, "")
+		cmd.Flags().String("display_name", mockChannelDisplayName, "")
+		cmd.Flags().String("team", mockTeam.Name, "")
+
+		mockChannel := &model.Channel{
+			Name:        mockChannelName,
+			DisplayName: mockChannelDisplayName,
+			TeamId: mockTeam.Id,
+			Type: model.CHANNEL_OPEN,
+		}
+
+		s.client.
+			EXPECT().
+			GetTeam(mockTeam.Name, "").
+			Return(mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			CreateChannel(mockChannel).
+			Return(mockChannel, &model.Response{Error: nil}).
+			Times(1)
+
+		err := createChannelCmdF(s.client, cmd, []string{})
+		s.Require().Nil(err)
+		s.Require().Equal(mockChannel, printer.GetLines()[0])
+		s.Require().Len(printer.GetLines(), 1)
+	})
+
+	s.Run("Create invite only channel", func() {
+		printer.Clean()
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", mockChannelName, "")
+		cmd.Flags().String("display_name", mockChannelDisplayName, "")
+		cmd.Flags().String("team", mockTeam.Name, "")
+		cmd.Flags().Bool("private", true, "")
+
+		mockChannel := &model.Channel{
+			Name:        mockChannelName,
+			DisplayName: mockChannelDisplayName,
+			TeamId: mockTeam.Id,
+			Type: model.CHANNEL_PRIVATE,
+		}
+
+		s.client.
+			EXPECT().
+			GetTeam(mockTeam.Name, "").
+			Return(mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			CreateChannel(mockChannel).
+			Return(mockChannel, &model.Response{Error: nil}).
+			Times(1)
+
+		err := createChannelCmdF(s.client, cmd, []string{})
+		s.Require().Nil(err)
+		s.Require().Equal(mockChannel, printer.GetLines()[0])
+		s.Require().Len(printer.GetLines(), 1)
+	})
```

Comment:
I miss here tests where the API calls for `GetTeam`fails, and one for the `CreateChannel` API call fails.

Reply:
@jespino by fail you mean return an error or when `GetTeam` returns a `nil`

Reply:
I mean when GetTeam returns an error.

Reply:
@jespino I might be wrong but the no error is returned by `getTeamFromTeamArg` and if I change it to return the error from `GetTeam` won't I break other parts of the code that use `getTeamFromTeamArg`

Reply:
Well... the underneath code is ignoring the error, so... yes, maybe makes no sense to test the error returned here. Actually I'm not very happy with how is handled the errors here, but that is another subject that we should discuss for other PR.

Reply:
@mgdelacroix What do you thing, we are swallowing the error here, should we raise the error to the interface some how, or at least leave the commands the possibility of handling the error differently than not-found teams?

Reply:
Totally agree. I can see the reasoning on why this was done at the beginning (to ease the use of the `getTeamsFromTeamArgs`) but I think this can be done better. Same happens in almost every `From*Args` function, so this may very well be a good refactor

Reply:
In any case, I'd say we should not tackle this change as part of the PR